src/components/App/App.jsx

```diff
@@ -1,60 +1,42 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
+import { closeModal, setMoviesThunk, setIsLoading } from 'store/actions';
+import { connect } from 'react-redux';
 import styles from './App.styl';
 import Header from '../Header/Header';
 import MovieHeader from '../MovieHeader/MovieHeader';
 import MainContent from '../MainContent/MainContent';
 import Footer from '../Footer/Footer';
 import WithWrapper from '../WithWrapper/WithWrapper';
-import fetchMovies from '../../utils/fetchMovies';
-import normalizeMoviesData from '../../utils/normalizeMoviesData';
-import MainContext from '../MainContext/MainContext';
 import ModalsWrapper from '../ModalsWrapper/ModalsWrapper';
-import { modalType } from '../Modal/Modal';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
 
 const MainContentWithWrapper = WithWrapper(MainContent);
 
-const App = () => {
-  const [moviesList, setMoviesList] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
-  const [currentModal, setCurrentModal] = useState(modalType.none);
-  const [isModalShown, setIsModalShown] = useState(false);
-
+const App = ({ closeModal, isModalShown, setMoviesThunk, setIsLoading }) => {
   const escapeHandler = e => e.which === 27 && isModalShown ? closeModal() : void 0;
 
-  const openModal = useCallback(modalType => {
-    setCurrentModal(modalType);
-    setIsModalShown(true);
-  }, [currentModal, isModalShown]);
-
-  const closeModal = useCallback(() => {
-    setCurrentModal(modalType.none);
-    setIsModalShown(false);
-  }, [currentModal, isModalShown]);
-
   useEffect(async () => {
-    const fetchedMoviesData = await fetchMovies();
-
+    setIsLoading(true);
+    await setMoviesThunk();
```

Comment:
Generally I agree, I just wanted to highlight that this function is async, regardless of what lib I'm using. So thunk here is used as term rather than a lib name